components/sup/src/manager/service.rs

```diff
@@ -168,10 +169,110 @@ impl TemplateUpdate {
 enum InitializationState {
     Uninitialized,
     Initializing,
+    InitializerFailed,
     InitializerFinished,
     Initialized,
 }
 
+#[derive(Debug)]
+pub struct PersistentServiceWrapper {
+    run_state: ServiceRunState,
+    inner:     Option<Service>,
+}
+
+#[derive(Debug, Copy, Clone, PartialEq, Eq)]
+pub enum RestartState {
+    None,
+    NeedsRestart,
+    Restarting,
+    Restarted,
+}
+
+impl Default for RestartState {
+    fn default() -> Self { RestartState::None }
+}
+
+#[derive(Debug, Default)]
+struct ServiceRunState {
+    ticks:            u64,
```

Comment:
doesn't seem like we ever use this. Do we need it?

Reply:
Yup, I've removed it